File: 1073-number-of-enclaves/number-of-enclaves.py
class Solution:
    def numEnclaves(self, grid: List[List[int]]) -> int:
        rows = len(grid)
        columns = len(grid[0])
        output = 0

        def isValid(i, j):
            if (i < 0 or j < 0 or i == rows or j == columns):
                return False
            return True

        def traverse(i, j):
            if (not isValid(i, j) or grid[i][j] != 1):
                return
            grid[i][j] = 0
            traverse(i - 1, j)
            traverse(i, j + 1)
            traverse(i + 1, j)
            traverse(i, j - 1)

        for i in range(rows):
            for j in range(columns):
                cell = grid[i][j]
                if (cell == 1):
                    if ((i == 0 or i == rows - 1) or (j == 0 or j == columns - 1)):
                        traverse(i, j)

        for i in range(rows):
            for j in range(columns):
                cell = grid[i][j]

                if (cell == 1):
                    output += 1


        return output
        

File: 1073-number-of-enclaves/number-of-enclaves.js
var numEnclaves = function(grid) {
    let rows = grid.length
    let columns = grid[0].length
    let count = 0

    function traverse(i, j) {
        if (i < 0 || j < 0 || i == rows || j == columns || grid[i][j] != 1) return
        grid[i][j] = 0
        traverse(i - 1, j) //up
        traverse(i, j + 1) //right
        traverse(i + 1, j) //down
        traverse(i, j - 1) //left
    }

    for (let i = 0; i < rows; i++) {
        for (let j = 0; j < columns; j++) {
            cell = grid[i][j]
            if (cell == 1) {
                if (i == 0 || j == 0 || i == rows - 1 || j == columns - 1) {
                    traverse(i, j)
                }
            }
        }
    }

    for (let i = 0; i < rows; i++) {
        for (let j = 0; j < columns; j++) {
            cell = grid[i][j]
            if (cell == 1) count += 1
        }
    }
    return count
};